File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_spaces(int spaces);
void print_blocks(int blocks);

int main(void)
{
    //Prompts the user for the height of the pyramid
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    int spaces = height - 1;
    int blocks = 1;
    for (int i = 0; i < height; i++)
    {
        //Prints the spaces for left pyramid
        print_spaces(spaces);
        //Prints the blocks for left pyramid
        print_blocks(blocks);

        printf("  "); //Prints the space between the pyramids

        //Prints the spaces for right pyramid
        print_blocks(blocks);

        spaces--;
        blocks++;
        printf("\n");
    }
}

void print_spaces(int spaces)
{
    for (int i = spaces; i > 0; i--)
    {
        printf(" ");
    }
}

void print_blocks(int blocks)
{
    for (int i = 0; i < blocks; i++)
    {
        printf("#");
    }
}

File: sentimental-mario-more/mario.py
# TODO
from cs50 import get_int

# Prompt user for the height of the pyramids (1 - 8 inclusive only)
while(True):
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Initialize the spaces before a block is printed
spaces = height - 1

# Go through every row of the pyramids
for row in range(height):
    # Print the appropriate spaces and decrement for the next row
    for space in range(spaces):
        print(" ", end="")
    spaces -= 1
    # Print the blocks for the first pyramid
    for i in range(row + 1):
        print("#", end="")
    # Print the two spaces between the pyramids
    print("  ", end="")
    # Print the second pyramid
    for i in range(row + 1):
        print("#", end="")
    # Print a new line
    print()